src/Microsoft.TestPlatform.CrossPlatEngine/Client/Parallel/ParallelProxyDiscoveryManager.cs

```diff
@@ -140,8 +140,10 @@ public bool HandlePartialDiscoveryComplete(IProxyDiscoveryManager proxyDiscovery
         var partiallyDiscoveredCount = _dataAggregator.GetSourcesWithStatus(DiscoveryStatus.PartiallyDiscovered).Count;
         var fullyDiscoveredCount = _dataAggregator.GetSourcesWithStatus(DiscoveryStatus.FullyDiscovered).Count;
         var expectedCount = _availableTestSources;
-        TPDebug.Assert(notDiscoveredCount + partiallyDiscoveredCount + fullyDiscoveredCount == expectedCount,
-            $"Total count of sources ({expectedCount}) should match the count of sources with status not discovered ({notDiscoveredCount}), partially discovered ({partiallyDiscoveredCount}) and fully discovered ({fullyDiscoveredCount}).");
+        // When this fails, look at the _dataAggregator and look at the sources that it holds. It is possible that adapter incorrectly reports
+        // the source on the testcase object. Each distinct source that will appear on TestCase will be considered a file.
+        //TPDebug.Assert(notDiscoveredCount + partiallyDiscoveredCount + fullyDiscoveredCount == expectedCount,
```

Comment:
Does removing this assert influence parallel discovery in any way ?

Reply:
I've put it back, it was broken, but fixed in another PR in the meantime.